unit_tests/test_scan_operation_thread.py

```diff
@@ -151,80 +153,37 @@ def test_negative_prometheus_validation_error(events, cluster):
                 ScanOperationThread(default_params)._run_next_operation()  # pylint: disable=protected-access
             all_events = get_event_log_file(events)
             assert "Severity.NORMAL" in all_events[0] and "period_type=begin" in all_events[0]
-            assert "Severity.ERROR" in all_events[1] and "period_type=end" in all_events[
+            severity = "Severity.ERROR"
+            if SkipPerIssues('https://github.com/scylladb/scylladb/issues/21578', TestConfig().tester_obj().params):
+                severity = "Severity.WARNING"
+            assert severity in all_events[1] and "period_type=end" in all_events[
                 1] and "Fullscan failed - 'mapreduce_service_requests_dispatched_to_other_nodes' was not triggered" in all_events[1]
 
 
-class ExecuteOperationTimedOutMockCqlConnectionPatient(MockCqlConnectionPatient):
-    def execute(*args, **kwargs):
-        # pylint: disable=unused-argument
-        # pylint: disable=no-method-argument
-        raise OperationTimedOut("timeout")
-
-
-class ExecuteAsyncOperationTimedOutMockCqlConnectionPatient(MockCqlConnectionPatient):
-    def execute_async(*args, **kwargs):
-        # pylint: disable=unused-argument
-        # pylint: disable=no-method-argument
-        raise OperationTimedOut("timeout")
-
-
-@pytest.mark.parametrize(("mode", 'severity', 'timeout', 'execute_mock'),
-                         [['partition', 'WARNING', 0, 'execute_async'],
-                          ['aggregate', 'ERROR', 60*30, 'execute'],
-                          ['table', 'WARNING', 0, 'execute']])
-def test_scan_negative_operation_timed_out(mode, severity, timeout, execute_mock, events, node):
+@pytest.mark.parametrize("exception", [ReadTimeout("Operation timed out"), Exception("Host has been marked down or removed"), OperationTimedOut("timeout")])
+@pytest.mark.parametrize("mode", ['table', 'partition', 'aggregate'])
+def test_scan_negative_execution_errors(mode, exception, events, node):
     # pylint: disable=redefined-outer-name
     # pylint: disable=too-many-arguments
-    if execute_mock == 'execute_async':
-        connection = ExecuteAsyncOperationTimedOutMockCqlConnectionPatient()
+    if mode == "partition":
+        class Connection(MockCqlConnectionPatient):
```

Comment:
Much better, but I would extract this class outside of the test:
```
class MockConnection(MockCqlConnectionPatient):
    def __init__(exception):
      self.exception = exception
    def execute(*args, **kwargs):
        # pylint: disable=unused-argument
        # pylint: disable=no-method-argument
        raise self.exception
    def execute_async(*args, **kwargs):
        raise self.exception
```

Reply:
for some reason joining into 1 class failed all partition test cases. 
```
FAILED unit_tests/test_scan_operation_thread.py::test_scan_negative_execution_errors[partition-exception2] - AssertionError: Subscriber <EventsFileLogger name='EventsFileLogger-2' pid=46 parent=7 started daemon> didn't receive 2 events in 10 seconds

```

I'm familiar with how  Mock event system  works

Reply:
pity, but extracting both of them outside would work right?

Reply:
Also no.

Reply:
Ok, no point in spending more time on it